projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/upgrade-plan-banner.jsx

```diff
@@ -32,19 +37,35 @@ const UpgradePlanBanner = ( {
 		onRedirect
 	);
 
+	const upgradeDescription = useSelect( select => {
+		if ( description ) {
+			return description;
+		}
+
+		const planSelector = select( 'wordpress-com/plans' );
+		const plan = planSelector && planSelector.getPlan( requiredPlan );
+		if ( plan ) {
+			return sprintf( UPGRADE_NUDGE_PLAN_DESCRIPTION, plan.product_name_short );
+		}
+
+		return null;
+	}, [] );
```

Comment:
I've noticed that many blocks and components use the compose function and place a withSelect inside instead of adding a useSelect call inside the actual component.
Just out of curiosity, could you explain why we are using useSelect in this case instead?

Reply:
`useSelect` and `withSelect` (should) work in the exact same way, but [hooks](https://reactjs.org/docs/hooks-intro.html) (`use*` functions) are less verbose, more readable, and just overall easier to use.

The performance difference is insignificant and the re-renders can be controlled by the dependencies array (the second argument).
Basically, the hook would only run when one of the variables listed as dependency changes.
In this case, the array is empty, meaning that the hook only runs _once_ with the initial value of all involved variables.

We should probably add `description` as dependency, to make sure the hook runs until it gains a value.